dotcom-rendering/src/web/components/ContainerTitle.tsx

```diff
@@ -53,18 +66,34 @@ const descriptionStyles = (fontColour?: string) => css`
 	}
 `;
 
+const dateTextStyles = (color: Colour) => css`
+	${headline.xxxsmall({ fontWeight: 'bold' })};
+	color: ${color};
+`;
+
+/**
+ * ContainerTitle
+ *
+ * For the date header to be shown, a valid editionId must be passed, as the
+ * date is based off of the edition timezone.
+ */
 export const ContainerTitle = ({
 	title,
 	fontColour,
 	description,
 	url,
-}: {
-	title?: string;
-	fontColour?: string;
-	description?: string;
-	url?: string;
-}) => {
+	containerPalette,
+	showDateHeader,
+	editionId,
+}: Props) => {
 	if (!title) return null;
+
+	const overrides =
+		containerPalette && decideContainerOverrides(containerPalette);
+
+	const now = new Date();
```

Comment:
Is there anywhere else we can extract this from, like the publication Date from one of the articles in the container? I imagine it would be... inconvenient, but I'm worried about this going out of sync with the articles in the container. My first thought was because of caching, but another more likely scenario is just that the container won't be updated with all new content at the stroke of (locale-specific) midnight, when this will change

Reply:
> but I'm worried about this going out of sync with the articles in the container

It's been a while since I worked on this exact logic, but I believe that this is the same way it currently works on Frontend, so let me double check!

Reply:
Searching through the frontend code I was able to confirm this - https://github.com/guardian/frontend/blob/main/common/app/views/fragments/containers/facia_cards/standardHeaderMeta.scala.html#L35

Reply:
There's definitely a margin for error here with the caching though - but I believe this is mostly used on network fronts which are updated very regularly!